.github/workflows/integration-test.yaml

```diff
@@ -30,5 +33,10 @@ jobs:
         uses: actions/checkout@v2
       - name: setup kind and run integration tests
         run: make integration-test
-      - name: cleanup all the kind clusters
+  cleanup:
+    runs-on: [self-hosted, aws-app-mesh-controller-for-k8s, X64]
+    if: ${{ always() }}
+    needs: [build]
+    steps:
+      - name: delete kind clusters
```

Comment:
Is it possible to have multiple test runs going at once? If so I would imagine this would break the other runs.

Reply:
@jeremymill @bendu do either of you have a definitive answer to joe's q?

Reply:
Only 1 job runs on the host at a time. They queue up if the host is busy